apps/wallet-mobile/src/features/Links/common/useLinksRequestAction.tsx

```diff
@@ -0,0 +1,101 @@
+import {linksCardanoModuleMaker, LinksTransferRequestAdaWithLinkParams, LinksYoroiAction, useLinks} from '@yoroi/links'
+import {useTransfer} from '@yoroi/transfer'
+import * as React from 'react'
+import {InteractionManager, Keyboard} from 'react-native'
+
+import {useModal} from '../../../components/Modal/ModalContext'
+import {useSelectedWalletContext} from '../../../SelectedWallet'
+import {Logger} from '../../../yoroi-wallets/logging'
+import {asQuantity, Quantities} from '../../../yoroi-wallets/utils/utils'
+import {RequestedAdaPaymentWithLinkScreen} from '../useCases/RequestedAdaPaymentWithLinkScreen/RequestedAdaPaymentWithLinkScreen'
+import {useNavigateTo} from './useNavigationTo'
+import {useStrings} from './useStrings'
+
+const heightBreakpoint = 467
+export const useLinksRequestAction = () => {
+  const strings = useStrings()
+  const {action, actionFinished} = useLinks()
+  const {openModal, closeModal} = useModal()
+  const [wallet] = useSelectedWalletContext()
+  const navigateTo = useNavigateTo()
+
+  const {memoChanged, receiverResolveChanged, amountChanged, reset, redirectToChanged} = useTransfer()
+  const startTransferWithLink = React.useCallback(
+    (action: LinksYoroiAction, decimals: number) => {
+      Logger.debug('startTransferWithLink', action, decimals)
+      if (action.info.useCase === 'request/ada-with-link') {
+        reset()
+        try {
+          const link = decodeURIComponent(action.info.params.link)
+          const parsedCardanoLink = linksCardanoModuleMaker().parse(link)
+          if (parsedCardanoLink) {
+            const redirectTo = action.info.params.redirectTo
+            if (redirectTo != null) redirectToChanged(decodeURIComponent(redirectTo))
+
+            const {address: receiver, amount, memo} = parsedCardanoLink.params
+            const ptAmount = Quantities.integer(asQuantity(amount ?? 0), decimals)
+            memoChanged(memo ?? '')
+            receiverResolveChanged(receiver ?? '')
+            amountChanged(ptAmount)
+            closeModal()
+            actionFinished()
+            navigateTo.startTransfer()
```

Comment:
@michaeljscript not long time ago it was decided to move away from params, cuz they aren't safe, it always need checking again on the Screens upon receiving them, that's why we favour context over params, yet, when exposing more routes to the word, we won't be able to avoid them anymore.